controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -52,11 +73,229 @@ func (r *StorageConsumerReconciler) Reconcile(ctx context.Context, request recon
+	}
+
+	r.storageConsumer.Status.State = v1alpha1.StorageConsumerStateReady
+	r.storageConsumer.Status.GrantedCapacity = r.storageConsumer.Spec.Capacity
 
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageConsumerReconciler) reconcileCephBlockPool() error {
+
+	failureDomain, err := r.getFailureDomain()
+	if err != nil {
+		return fmt.Errorf("Error getting failure domain. %+v", err)
+	}
+
+	cephBlockPool := r.newCephBlockPool(failureDomain)
+
+	_, err = ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephBlockPool, func() error {
+		if err := r.own(r.cephBlockPool); err != nil {
+			return err
+		}
+
+		r.cephBlockPool.Spec = cephBlockPool.Spec
+		return nil
+	})
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephBlockPool.", "CephBlockPool", klog.KRef(r.cephBlockPool.Namespace, r.cephBlockPool.Name))
+		return err
+	}
+
+	r.storageConsumer.Status.CephObjects.BlockPoolName = r.cephBlockPool.Name
+	return nil
+}
+
+func (r *StorageConsumerReconciler) reconcileCephClientRbd() error {
+
+	cephClientRBD := r.newCephClientRBD()
+
+	_, err := ctrl.CreateOrUpdate(r.ctx, r.Client, r.cephClientRBD, func() error {
+		if err := r.own(r.cephClientRBD); err != nil {
+			return err
+		}
+
+		r.cephClientRBD.Spec = cephClientRBD.Spec
+		return nil
+	})
+
+	if err != nil {
+		r.Log.Error(err, "Failed to update CephClient.", "CephClient", klog.KRef(r.cephClientRBD.Namespace, r.cephClientRBD.Name))
+		return err
+	}
+
+	r.storageConsumer.Status.CephObjects.CephRbdUser = r.cephClientRBD.Name
+	return nil
+}
+
+func (r *StorageConsumerReconciler) newCephBlockPool(failureDomain string) *rookCephv1.CephBlockPool {
+
+	capacity := r.storageConsumer.Spec.Capacity.String()
+
+	return &rookCephv1.CephBlockPool{
+		Spec: rookCephv1.PoolSpec{
+			FailureDomain: failureDomain,
+			Replicated: rookCephv1.ReplicatedSpec{
+				Size:                     3,
+				ReplicasPerFailureDomain: 1,
+				TargetSizeRatio:          .49,
+			},
+			EnableRBDStats: true,
+			Quotas: rookCephv1.QuotaSpec{
+				MaxSize: &capacity,
+			},
+		},
+	}
+}
+
+func (r *StorageConsumerReconciler) newCephClientRBD() *rookCephv1.CephClient {
+
+	return &rookCephv1.CephClient{
+		Spec: rookCephv1.ClientSpec{
+			Caps: map[string]string{
+				"mon": "profile rbd",
+				"osd": fmt.Sprintf("profile rbd pool=%s", r.cephBlockPool.Name),
+			},
+		},
+	}
+}
+
+func (r *StorageConsumerReconciler) areRookCephResourcesReady() bool {
+	// Check status for cephBlockPool resource.
+	if err := r.get(r.cephBlockPool); err == nil {
+		if r.cephBlockPool.Status.Phase != "Ready" {
+			return false
+		}
+	} else {
+		return false
+	}
+
+	// Check status for cephClient resource of rbd.
+	if err := r.get(r.cephClientRBD); err == nil {
+		if r.cephClientRBD.Status.Phase != "Ready" {
+			return false
+		}
+	} else {
+		return false
+	}
+	return true
+}
+
+func (r *StorageConsumerReconciler) get(obj client.Object) error {
+	key := client.ObjectKeyFromObject(obj)
+	return r.Client.Get(r.ctx, key, obj)
+}
```

Comment:
What benefit does this wrapper provide?

Reply:
`get(object)` -> `get(ctx, key , obj)`
The wrapper auto-populates the additional info from the object itself and the reconciler, this way we do not need to do it on every call site